hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/validate/HazelcastSqlValidator.java

```diff
@@ -117,7 +119,7 @@ protected void addToSelectList(
             SelectScope scope,
             boolean includeSystemVars
     ) {
-        if (isHiddenColumn(exp, scope)) {
+        if (isHiddenColumn(exp, scope) && !isUpdate) {
```

Comment:
Why do we need this? I tried to remove it and all tests in `SqlUpdateTest` pass. Isn't it a leftover?

Reply:
Unfortunately, it's not a leftover. Calcite expands `*` in couple of places during `UPDATE`. If we omit hidden fields we end up [here](https://github.com/apache/calcite/blob/03e356c656c2bc98b1a273352475033545e0928d/core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java#L2002) - so I would say it works purely by accident.
On the other hand I wasn't able to devise a proper test in which we could validate `isUpdate` fulfils it's purpose...

Reply:
I don't understand. How did you notice the issue then when everything works without it? 
We should at least add some explanatory comment because I see myself trying to remove it in a year :)